### PR TITLE
Get rid of KPR=probe and socket-LB protocols

### LINKED ISSUE #22076: agent: Remove KPR=probe
The option has been deprecated, and it can be safely removed in v1.13.

### LINKED ISSUE #22077: Remove host-reachable svc leftovers
From the upgrade guide:

> The enable-host-reachable-services option (.hostServices in Helm) was renamed to bpf-lb-sock (.socketLB in Helm), and will be removed in version 1.13.

> The host-reachable-services-protos option (.hostServices.protocols in Helm) was deprecated, and it will be removed in version 1.13.

### PR DESCRIPTION
**Pull request by brb**
Both were deprecated in v1.12, and were scheduled for removal in v1.13.

Fix #22076
Fix #22077

### CODE REVIEW

**Review by tklauser — APPROVED** [review]

**Review by gandro — APPROVED** [review]
Agent and Helm looks good to me!

File: pkg/option/config.go [hunk 1]
```diff
@@ -1603,9 +1587,7 @@ type DaemonConfig struct {
 	DebugVerbose                  []string
 	EnableSocketLB                bool
 	EnableSocketLBTracing         bool
-	EnableHostServicesTCP         bool
-	EnableHostServicesUDP         bool
-	EnableHostServicesPeer        bool
+	EnableHostServicesPeer        bool // TODO(brb) rename to SocketLBPeer
```
Comment (ti-mo) [inline pkg/option/config.go]: What's preventing the rename?

**Review by ti-mo — COMMENTED** [review]
Nice. `init.sh` removals always welcome! One minor comment.

Heads up @rgo3

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -320,6 +320,10 @@ Removed Options
 
 * The ineffective ``disable-conntrack``, ``endpoint-interface-name-prefix`` options deprecated in
   version 1.12 have been removed.
+* The ``host-reachable-services-protos`` option deprecated in version v1.12 has
+  been removed.
+* The ``probe`` option of ``kube-proxy-replacement`` deprecated in version v1.12
+  has been removed.
```
Comment (borkmann) [inline Documentation/operations/upgrade.rst]: What is the recommendation we provide to folks who have been running probe before? We should probably provide a short blurb in here, so steps are clear when they upgrade.

**Review by borkmann — APPROVED** [review]

File: daemon/cmd/kube_proxy_replacement.go [hunk 3]
```diff
@@ -70,21 +64,16 @@ func initKubeProxyReplacementOptions() (bool, error) {
...
 	}
 
-	// strict denotes to panic if any to-be enabled feature cannot be enabled
-	strict := option.Config.KubeProxyReplacement != option.KubeProxyReplacementProbe
-
-	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementProbe ||
-		option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict {
-
-		log.Infof("Trying to auto-enable %q, %q, %q, %q, %q features",
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict {
```
Comment (pchaigno) [inline daemon/cmd/kube_proxy_replacement.go:73]: As a follow up, would it also make sense to rename `strict` to `all`? Now it's `strict` vs. `partial`/`disabled` which doesn't make a whole lot of sense IMO. It made sense when defined against `probe`.

**Review by pchaigno — APPROVED** [review]

**Comment by brb** [comment]
The `ci-external` failure is legit:
```
Error: unknown flag: --enable-host-reachable-services
```
This is coming from https://github.com/cilium/cilium-cli/blob/master/clustermesh/clustermesh.go#L1485 :disappointed:

File: pkg/option/config.go (hunk 1 above)
Comment (brb) [reply pkg/option/config.go]: Oops, good catch.

File: daemon/cmd/kube_proxy_replacement.go (hunk 3 above)
Comment (brb) [reply daemon/cmd/kube_proxy_replacement.go:73]: Good point. I was thinking about taking it even further:
- `strict`: `--enable-kpr`.
- `partial / disabled`: controlled via individuals flags (e.g., `--enable-node-port`).

However, to get things rolling, I'd do in a separate PR after the v1.13 release.

File: Documentation/operations/upgrade.rst (hunk 2 above)
Comment (brb) [reply Documentation/operations/upgrade.rst]: Aye, good idea. Will add some.

**Comment by brb** [comment]
The Jenkins CI hit the IPSec flakes, while `ci-multicluster` and `ci-external-workloads` are waiting for https://github.com/cilium/cilium/pull/22140.

**Review by nathanjsweet — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

File: .github/workflows/conformance-externalworkloads.yaml [hunk 4]
```diff
@@ -68,7 +68,7 @@ env:
   vmName: ${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}-vm
   zone: us-west2-a
   vmStartupScript: .github/gcp-vm-startup.sh
-  cilium_cli_version: v0.12.4
+  cilium_cli_version: v0.12.7
```
Comment (pchaigno) [inline .github/workflows/conformance-externalworkloads.yaml:71]: Why didn't you update other workflows?

Comment (brb) [reply .github/workflows/conformance-externalworkloads.yaml:71]: Just to check whether the previous failure was fixed. https://github.com/cilium/cilium/pull/22140 is supposed to update all workflows.

**Pull request merged**